app/src/main/res/layout/fragment_product_details_page.xml

```diff
@@ -0,0 +1,127 @@
+        tools:context=".presentation_layer.mainScreen.main_fragments.pdp.ProductDetailsPageFragment">
+
+        <androidx.appcompat.widget.AppCompatImageView
+            android:id="@+id/product_image"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:contentDescription="@{null}"
+            app:imageUrl="@{viewModel.product.imageURL}" />
+
+        <androidx.appcompat.widget.AppCompatButton
+            android:id="@+id/add_to_cart_button"
+            style="@style/addToCartButtonStyle"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/margin32"
+            android:layout_marginEnd="@dimen/margin32"
+            android:layout_marginBottom="@dimen/margin32"
+            android:text="@string/add_to_cart"
+            app:layout_constraintBottom_toBottomOf="@id/product_view"
+            app:layout_constraintEnd_toEndOf="@id/product_view"
+            app:layout_constraintStart_toStartOf="@id/product_view" />
+
+        <View
+            android:id="@+id/product_view"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:background="@drawable/product_details_view" />
+
+        <androidx.appcompat.widget.AppCompatTextView
+            android:id="@+id/price_text"
+            style="@style/priceText"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/margin32"
+            android:layout_marginEnd="@dimen/margin32"
+            app:layout_constraintEnd_toEndOf="@id/product_view"
+            app:layout_constraintTop_toTopOf="@id/product_view"
+            app:setPrice="@{viewModel.product.price}" />
+
+        <androidx.appcompat.widget.AppCompatTextView
+            android:id="@+id/title_text"
+            style="@style/productTitleText"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/margin32"
+            android:layout_marginTop="@dimen/margin32"
+            android:layout_marginEnd="@dimen/margin23"
+            android:text="@{viewModel.product.title}"
+            app:layout_constraintEnd_toStartOf="@id/price_text"
+            app:layout_constraintStart_toStartOf="@id/product_view"
+            app:layout_constraintTop_toTopOf="@id/product_view" />
+
+        <androidx.appcompat.widget.AppCompatRatingBar
+            android:id="@+id/rating_bar"
+            style="?android:attr/ratingBarStyleSmall"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/margin32"
+            android:layout_marginTop="@dimen/margin12"
+            android:numStars="5"
+            android:rating="@{viewModel.product.rating.rate}"
+            android:stepSize="0.1"
```

Comment:
may be is it better to set it in your style.xml?

Reply:
done